Cannot compile opencv_contrib 4.0.0

VS 2017 x64 windows 10 64 bits
cmake 3.11.0
opencv (Merge pull request #12467 from alalek:core_use_shared_ptr) and opencv_contrib  (Merge pull request #1765 from alalek:core_use_shared_ptr)

Error in correspondance project 
```
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\nrobustviewmatching.cc(156): error C2088: '==': illegal for struct
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\nrobustviewmatching.cc(183): error C2088: '==': illegal for struct
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\nrobustviewmatching.cc(156): error C2666: 'std::operator ==': 3 overloads have similar conversions
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\nrobustviewmatching.cc(183): error C2666: 'std::operator ==': 3 overloads have similar conversions

```
replaced cv::Ptr with std::shared_ptr [here ](https://github.com/opencv/opencv_contrib/blob/master/modules/sfm/src/libmv_light/libmv/correspondence/nRobustViewMatching.h#L130-L132)solves problem

--- a/modules/sfm/src/libmv_light/libmv/correspondence/nRobustViewMatching.h
+++ b/modules/sfm/src/libmv_light/libmv/correspondence/nRobustViewMatching.h
@@ -127,9 +127,9 @@ private :
   Matches m_tracks;
 
   /// Interface to detect Keypoint.
-  cv::Ptr<cv::FeatureDetector> m_pDetector;
+  std::shared_ptr<cv::FeatureDetector> m_pDetector;
   /// Interface to describe Keypoint.
-  cv::Ptr<cv::DescriptorExtractor> m_pDescriber;
+  std::shared_ptr<cv::DescriptorExtractor> m_pDescriber;
 };
 
 } // using namespace correspondence

